addons/mail/static/src/models/thread/thread.js

```diff
@@ -607,6 +607,36 @@ function factory(dependencies) {
             this.fetchAndUpdateSuggestedRecipients();
         }
 
+        /**
+         * Fetch messages that contains searched keyword. Useful when messages are
+         * not loaded in the chatter.
+         */
+        async getMessages() {
+            this.update({ isSearchingMessages: true });
+            const domain = [
+                '&','&','&',
+                ['res_id', '=', this.id],
+                ['model', '=', this.model],
+                ['id', 'not in', this.messages.map(message => message.id)],
+                '|',
+                ['body', 'ilike', this.searchedText],
+                ['subtype_id.description', 'ilike', this.searchedText],
+            ];
+            const messagesData = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'get_messages',
+                kwargs: { domain },
+            }));
+            this.update({
+                filteredMessages: [['insert',
+                    messagesData.map(data =>
+                        this.env.models['mail.message'].convertData(data)
+                    )
+                ]],
+                isSearchingMessages: false,
+            });
```

Comment:
You are re-implementing thread cache here, we already have custom message_fetch with domain for search in discuss. All you have to do when there is a search in chatter in to change thread cache from main cache (no domain) to thread cache with search domain.

Reply:
Yes! we already have `message_fetch` with domain for search. But it has a limit in arguments and it will return a limited amount of formatted messages with a provided domain. Here we want all matched ones.
What would you like to suggest? either make a new function or just pass one more argument for `message_fetch`

Reply:
I don't think we want all matched one. If it's a very long thread, and there are hundreds of search matches, we shouldn't return them all at the same time.